lib/generator.rb

```diff
@@ -36,12 +37,24 @@ def test_cases
   end
 
   def generate
+    generate_test_file
+    update_version
+  end
+
+  private
+
+  def generate_test_file
     File.open(path_to("#{name.gsub(/[ -]/, '_')}_test.rb"), 'w') do |f|
       f.write ERB.new(File.read(path_to('example.tt'))).result binding
+      FileUtils.chmod('+x', f)
```

Comment:
Is this available in all languages?  We should leave the file executable bit to the helper tool in bin.  It is outside the concern of the generator.